elements/standard/bwratedunqueue.cc

```diff
@@ -19,13 +19,28 @@
  */
 
 #include <click/config.h>
+#include <click/args.hh>
 #include "bwratedunqueue.hh"
 CLICK_DECLS
 
-BandwidthRatedUnqueue::BandwidthRatedUnqueue()
+BandwidthRatedUnqueue::BandwidthRatedUnqueue() : _use_extra_length(false), _link_rate(false)
 {
 }
 
+int
+BandwidthRatedUnqueue::configure(Vector<String> &conf, ErrorHandler *errh)
+{
+    if (Args(this, errh).bind(conf)
+        .read_or_set("EXTRA_LENGTH", _use_extra_length, false)
+        .read_or_set("LINK_RATE", _link_rate, false)
+        .consume() < 0)
+        return -1;
+    return RatedUnqueue::configure(conf, errh);
+}
+
+#define LENGTHOF(p) \
+    ((_link_rate?8:1)*(p->length() + _use_extra_length? EXTRA_LENGTH_ANNO(p) : 0) + (_link_rate? 8 * 24:0))
```

Comment:
I suppose that the purpose here is to convert into bits/sec over Ethernet when link_rate=true.

(_link_rate?8:1) converts a packet's length (in bytes) to bits.
(_link_rate? 8 * 24:0) is the sum of 1. Inter Frame Gap (12 bytes) + 2. MAC Preamble (8 bytes) + 3. CRC (4 bytes) = 24 bytes. We multiply by 8 to convert to bits and we are set.

Looks good to me :p
I just wanted to verify it analytically :)

Reply:
Yes you're right. It's not like if I messed it up in the past :p